crates/rpc/src/method/get_compiled_casm.rs

```diff
@@ -0,0 +1,1549 @@
+    Internal(anyhow::Error),
+}
+
+impl From<anyhow::Error> for Error {
+    fn from(error: anyhow::Error) -> Self {
+        Self::Internal(error)
+    }
+}
+
+impl From<Error> for crate::jsonrpc::RpcError {
+    fn from(error: Error) -> Self {
+        match error {
+            Error::CompilationFailed => Self::ApplicationError(ApplicationError::CompilationFailed),
+            Error::ClassHashNotFound(_) => {
+                Self::ApplicationError(ApplicationError::ClassHashNotFound)
+            }
+            Error::Internal(e) => Self::InternalError(e),
+        }
+    }
+}
+
+/// Get the compiled casm for a given class hash.
+pub async fn get_compiled_casm(context: RpcContext, input: Input) -> Result<Output, Error> {
+    let span = tracing::Span::current();
+    let jh = tokio::task::spawn_blocking(move || -> Result<Output, Error> {
+        let _g = span.enter();
+
+        let mut db = context
+            .storage
+            .connection()
+            .context("Opening database connection")
+            .map_err(Error::Internal)?;
+
+        let tx = db
+            .transaction()
+            .context("Creating database transaction")
+            .map_err(Error::Internal)?;
+
+        // Get the class definition
+        let casm_definition = tx
+            .casm_definition(input.class_hash)
+            .context("Fetching class definition")
+            .map_err(Error::Internal)?
+            .ok_or(Error::ClassHashNotFound(input.class_hash))?;
+
+        // Convert to JSON string
+        let casm_definition_str = String::from_utf8_lossy(&casm_definition);
+
+        // Parse the casm definition
+        let casm_contract_class = CasmContractClass::try_from(casm_definition_str.as_ref())
+            .context("Parsing casm definition")
+            .map_err(|_| Error::CompilationFailed)?;
+
+        Ok(Output(casm_contract_class))
+    });
+
+    jh.await.context("Fetching compiled casm")?
+}
+
+#[cfg(test)]
+mod tests {
+
+    use pathfinder_common::casm_class::{
+        BigUintAsHex,
+        CasmContractEntryPoint,
+        CasmContractEntryPoints,
+    };
+    use pathfinder_common::{casm_hash, class_hash, felt, sierra_hash, EntryPoint};
+    use serde_json::json;
+    use starknet_gateway_test_fixtures::class_definitions::{
+        CAIRO_1_1_0_BALANCE_CASM_JSON,
+        CAIRO_1_1_0_BALANCE_SIERRA_JSON,
+    };
+    use tempfile::tempdir;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_get_casm() {
+        let rpc = setup().await.expect("RPC context");
+
+        // Test the happy path
+        let result = get_compiled_casm(rpc.clone(), input())
+            .await
+            .expect("result");
+        assert_eq!(result.0, expected());
+
+        // Test the not found path
+        let result = get_compiled_casm(rpc, input_not_found())
+            .await
+            .expect_err("result");
+        assert_matches::assert_matches!(result, Error::ClassHashNotFound(_));
+    }
```

Comment:
I usually just prefer having these two cases as separate `#[test]`s so that when there's a failure you immediately know what failed just based on the name of the test. It's a matter of preference -- but we use that pattern throughout our RPC tests.

Reply:
That's a great point 👍